org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/translator/A4Solution.java

```diff
@@ -1315,26 +1569,17 @@ A4Solution solve(final A4Reporter rep, Command cmd, Simplifier simp, boolean try
             return this;
         // If cmd==null, then all four arguments are ignored, and we simply use
         // the lower bound of each relation
+        // [HASLab] refactored into A4Solution#solve(A4Reporter,A4Solution)
         if (cmd == null) {
-            Instance inst = new Instance(bounds.universe());
-            for (int max = max(), i = min(); i <= max; i++) {
-                Tuple it = factory.tuple("" + i);
-                inst.add(i, factory.range(it, it));
-            }
-            for (Relation r : bounds.relations())
-                inst.add(r, bounds.lowerBound(r));
-            eval = new Evaluator(inst, solver.options());
-            rename(this, null, null, new UniqueNameGenerator());
-            solved();
-            return this;
+            throw new RuntimeException("Should not be null, refactored.");
```

Comment:
`cmd == null` was used to create solutions from the static reader; this has been refactored as a different method